hypothesis-python/src/hypothesis/strategies.py

```diff
@@ -1575,10 +1575,10 @@ def __init__(self, values):
 
     def do_draw(self, data):
         # Reversed Fisher-Yates shuffle. Reverse order so that it shrinks
-        # propertly: This way we prefer things that are lexicographically
+        # properly: This way we prefer things that are lexicographically
         # closer to the identity.
         result = list(self.values)
-        for i in hrange(len(result)):
+        for i in hrange(len(result) - 1):
```

Comment:
I know there's an explanation in the commit message, but it would be nice to have a comment here too, e.g.:

```python
# It's safe to stop iterating just before the final element,
# because it would only be able to swap with itself, which
# is a no-op.
```